egs/multi_lang/s5/local/nnet3/run_tdnn_joint_babel_sp_bnf.sh

```diff
@@ -0,0 +1,257 @@
+  # we do this as it's a convenient way to get the stats for the 'lda-like'
+  # transform.
+  $cmd $dir/log/nnet_init.log \
+    nnet3-init --srand=-2 $dir/configs/init.config $dir/init.raw || exit 1;
+fi
+
+. $dir/configs/vars || exit 1;
+
+if [ $stage -le 10 ]; then
+  for lang in `seq 0 $[$num_lang-1]`; do
+    egs_dir=${multi_egs_dirs[$lang]}
+    ali_dir=${multi_ali_dirs[$lang]}
+    data=data/${lang_list[$lang]}/train${suffix}${aux_suffix}
+    num_frames=$(steps/nnet2/get_num_frames.sh $data)
+    echo num_frames = $num_frames
+    # sets samples_per_iter to have approximately 
+    # same number of archives per language.
+    samples_per_iter=$[$num_frames/($avg_num_archives*$frames_per_eg)]
+    online_ivector_dir=
+    if $use_ivector; then
+      online_ivector_dir=exp/${lang_list[$lang]}/nnet3/ivectors_train${suffix}${ivector_suffix}
+    fi
+    if [ ! -d "$egs_dir" ]; then
+      echo "$0: Generate egs for ${lang_list[$lang]}"
+      if [[ $(hostname -f) == *.clsp.jhu.edu ]] && [ ! -d $egs_dir/storage ]; then
+        utils/create_split_dir.pl \
+         /export/b0{3,4,5,6}/$USER/kaldi-data/egs/${lang_list[$lang]}-$(date +'%m_%d_%H_%M')/s5/$egs_dir/storage $egs_dir/storage
+      fi
+
+      extra_opts=()
+      [ ! -z "$cmvn_opts" ] && extra_opts+=(--cmvn-opts "$cmvn_opts")
+      [ ! -z "$feat_type" ] && extra_opts+=(--feat-type $feat_type)
+      [ ! -z "$online_ivector_dir" ] && extra_opts+=(--online-ivector-dir $online_ivector_dir)
+      extra_opts+=(--left-context $left_context)
+      extra_opts+=(--right-context $right_context)
+      echo "$0: calling get_egs.sh"
+      steps/nnet3/get_egs.sh $egs_opts "${extra_opts[@]}" \
+          --samples-per-iter $samples_per_iter --stage $get_egs_stage \
+          --cmd "$cmd" $egs_opts \
+          --frames-per-eg $frames_per_eg \
+          $data $ali_dir $egs_dir || exit 1;
+      
+    fi
+  done
+fi
+echo print-interval = $print_interval
+if [ $stage -le 11 ]; then
+  echo "$0: training mutilingual model."
+  steps/nnet3/multi/train_tdnn.sh --cmd "$train_cmd" \
+  --use-ivector $use_ivector --print-interval $print_interval \
+  --separate-init-layer-per-output $use_sep_init_layer \
+  --num-epochs $num_epochs --cleanup false \
+  --num-jobs-initial $num_jobs_initial --num-jobs-final $num_jobs_final \
+  --stage $train_stage \
+  --initial-effective-lrate $init_lrate --final-effective-lrate $final_lrate \
+  "${lang_list[@]}" "${multi_ali_dirs[@]}" "${multi_egs_dirs[@]}" \
+  $dir || exit 1;
+
+fi
+
+# decoding different languages
+if [ $stage -le 12 ]; then
+  num_decode_lang=${#decode_lang_list[@]}
+  (
+  for lang in `seq 0 $[$num_decode_lang-1]`; do
+    if [ ! -f $dir/${decode_lang_list[$lang]}/decode_dev10h.pem/.done ]; then 
+      cp $dir/cmvn_opts $dir/${decode_lang_list[$lang]}/.
+      echo decoding lang ${decode_lang_list[$lang]} using multilingual model $dir
+      run-4-anydecode-langs.sh --use-sep-init-layer $use_sep_init_layer \
```

Comment:
I don't think it is standard to call a main run script from a local. You can may be leave this as a comment.